docs/articles/documentation/extending-testcafe/custom-reporter/creating-reporter-plugin/helper-methods.md

```diff
@@ -0,0 +1,297 @@
+escapeHtml (str) → this
+```
+
+Parameter   | Type   | Description
+----------- | ------ | ------------------------------
+`str`       | String | The string you want to escape.
+
+**Example**
+
+```js
+reportFixtureStart (name) {
+    this.currentFixtureName = this.escapeHtml(name);
+}
+```
+
+### formatError
+
+Returns the formatted error message string for the specified error.
+
+```text
+formatError (err, prefix = '') → this
+```
+
+Parameter     | Type   | Description
+------------- | ------ | -------------------------------------------------------------------
+`err`         | String | The error message you need to format.
+`prefix = ''` | String | The string that is prepended to the error. By default, it is empty.
+
+**Example**
+
+The following example demonstrates how to add numbering to the errors.
+
+```js
+reportTestDone (name, errs){
+    ...
+    var hasErr = !!errs.length;
+
+    if (hasErr) {
+        errs.forEach((err, idx) => {
+            this.newline()
+                .write(this.formatError(err, `${idx + 1}) `))
+        });
+    }
+}
+```
+
+## Output Helpers
+
+### newline
+
+Adds a new line to the report.
+
+```text
+newline () → this
+```
+
+**Example**
+
+```js
+reportTaskDone (endTime, passed) {
+    ...
+    var summary = passed === this.testCount ?
+                  `${this.testCount} passed` :
+                  `${this.testCount - passed}/${this.testCount} failed`;
+
+    this.newline()
+        .write(summary)
+        .newline();
+}
+```
+
+### write
+
+Writes the specified text in the report.
+
+```text
+write (text) → this
+```
+
+Parameter | Type   | Description
+--------- | ------ | ---------------------------------
+`text`    | String | The text to output in the report.
+
+**Example**
+
+```js
+reportTaskDone (endTime, passed) {
+    ...
+    var summary = passed === this.testCount ?
+                  `${this.testCount} passed` :
+                  `${this.testCount - passed}/${this.testCount} failed`;
+
+    this.write(summary)
+        .newline();
+}
+```
+
+### useWordWrap
+
+Toggles word wrapping for subsequent output.
+
+```text
+useWordWrap (use) → this
+```
+
+Parameter | Type    | Description
+--------- | ------- | ---------------------------------------
+`use`     | Boolean | Specify *true* to enable word wrapping.
+
+**Example**
+
+The following example demonstrates how to enable word wrapping in case when the `title` string might be long.
+
+```js
+reportTestDone (name, errs) {
+    ...
+    name = `${this.currentFixtureName} - ${name}`;
+
+    var title = `${result} ${name}`;
+
+    this.useWordWrap(true)
+        .write(title);
+}
+```
+
+### setIndent
+
+Specifies indentation for subsequent output.
+
+```text
+setIndent (val) → this
+```
+
+Parameter | Type    | Description
+--------- | ------- | ----------------------------------------------------
+`val`     | Integer | Specifies the number of spaces to indent a new line.
+          |         | To disable indentation, set the parameter to 0.
```

Comment:
Don't place second sentence to the newline. it recognized by markdown as an additional row